### ISSUE TITLE
LRP returns negative values for a simple CNN model. 

### ISSUE THREAD

**Issue by ShihengDuan** [pytorch/captum#765]
Hi all,

Thanks for the support of Layer wise propagation. I have a simple CNN model with several convolutional and average pooling layers. From my understanding, LRP should always return positive values and the summation from all pixels should be 1. But I'm getting a negative summation and also lots of negative points. What does the negative value mean? How should I fix it? 

Thanks! 

```
class CustomModel(nn.Module):
    
    def __init__(self, model) -> None:
        super(CustomModel, self).__init__()
        self.model = model

    def forward(self, psl):
        psl = psl.reshape(psl.shape[0], psl.shape[1], psl.shape[2], psl.shape[3], n_vars)
        psl = psl[:,:-1]
        out = self.model(psl)
        return out        
custom_model = CustomModel(model)

lrp = LRP(custom_model)

lrps = []
for ind in tqdm(range(test_ds.__len__())):
    sample_lrp = lrp.attribute(test_ds[ind][0].reshape(1, 2, 192, 288, 3).to(device)).to('cpu').data.numpy()
    lrps.append(sample_lrp)
```

**Comment by ShihengDuan** [comment]
```model``` in CustomModel is my CNN model and since the input needs some transformation, I wrapped it into another model.

**Comment by bilalsal** [comment]
Hi @ShihengDuan,

Is is possible to construct a test case that does not rely on the wrapper model, just to rule out the possibility that input reshaping is causing the negative attributions?

@nanohanno Did you also encounter negative values when using LRP?

**Comment by ShihengDuan** [comment]
Hi @bilalsal 

Thanks for your reply. I have tried the following but it still gave me a negative attribution. 

```
lrp = LRP(model)
psl = test_ds[ind][0].reshape(1, 2, 192, 288, 3).to(device)
psl = psl.reshape(psl.shape[0], psl.shape[1], psl.shape[2], psl.shape[3], n_vars)
psl = psl[:,:-1]
sample_lrp = lrp.attribute(psl).to('cpu').data.numpy()
print(sample_lrp.shape)
print(np.min(sample_lrp))
print(np.sum(sample_lrp))
```

Both the min and sum return negative values.

**Comment by ShihengDuan** [comment]
Hi @NarineK, sorry for the misunderstanding of LRP. I see the LRP can be negative in the original paper. When I set ```return_convergence_delta=True```, I checked the sum of LRP and delta equals to the model prediction. Thanks for pointing out that. 

But from the documentation of LRP, the output attribution should already be normalized by the output and the summation of LRP should be 1. It seems that from the calculation, we are getting the original score rather than the normalized version. Please see below that the summation of LRP is not 1. In this case, should we divide it by the model output and get the normalized LRP? Is that correct? 

Thanks! 

```
delta:  tensor([-0.0229], device='cuda:0', grad_fn=<SubBackward0>)
sum of lrp:  tensor(-0.1870, device='cuda:0', grad_fn=<SumBackward0>)
output_true:  tensor([0.3936])
output_pred:  tensor([[-0.2099]], grad_fn=<CopyBackwards>)
sum of lrp + delta:  tensor([-0.2099], device='cuda:0', grad_fn=<AddBackward0>)
```

**Comment by NarineK** [comment]
@ShihengDuan, I think that that doc might be outdated. Because of the consistency reasons with other attribution methods we decided to multiply by output and the attributions are no longer normalized . See line:
https://github.com/pytorch/captum/blob/master/captum/attr/_core/lrp.py#L203  

Thus, the sum of the attributions is equal to the output, hence the delta ~= 0

**Comment by ShihengDuan** [comment]
@NarineK Ok then it makes sense to me. Thank you!

**Comment by ShihengDuan** [comment]
It might be worth updating the documentation.

**Comment by nanohanno** [comment]
That seems to be true, the documentation in appears to be outdated: https://github.com/pytorch/captum/blob/e1575c52f39e403f71cfda412b758df34d857914/captum/attr/_core/lrp.py#L151-L154 Sorry for not being very responsive atm. 😮‍💨
